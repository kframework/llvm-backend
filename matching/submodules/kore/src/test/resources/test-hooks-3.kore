[]

module TEST-HOOKSYMBOL-1

  sort S{} []

  hooked-symbol f{}() : S{} []

endmodule []
