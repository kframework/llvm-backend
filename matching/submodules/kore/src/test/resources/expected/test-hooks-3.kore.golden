Definition
    { definitionAttributes = Attributes []
    , definitionModules =
        [ Module
            { moduleName = ModuleName "TEST-HOOKSYMBOL-1"
            , moduleSentences =
                [ ObjectSentence (SentenceSortSentence SentenceSort
                    { sentenceSortName = (Id "S" AstLocationNone) :: Id Object
                    , sentenceSortParameters = []
                    , sentenceSortAttributes = Attributes []
                    })
                , ObjectSentence (SentenceHookSentence (SentenceHookedSymbol SentenceSymbol
                    { sentenceSymbolSymbol =
                        Symbol
                            { symbolConstructor = (Id "f" AstLocationNone) :: Id Object
                            , symbolParams = []
                            }
                    , sentenceSymbolSorts = []
                    , sentenceSymbolReturnSort =
                        SortActualSort SortActual
                            { sortActualName = (Id "S" AstLocationNone) :: Id Object
                            , sortActualSorts = []
                            }
                    , sentenceSymbolAttributes = Attributes []
                    }))
                ]
            , moduleAttributes = Attributes []
            }
        ]
    }